### PR TITLE
Add docs for Avatar, Icon; Reformat Badge docs

### PR DESCRIPTION
**Pull request by martinxu9**
## Summary
- Docs for avatar, icon
- Rearrange Badge docs per https://www.notion.so/reflex-dev/Usage-Low-Level-c2ecd6baf14f461c8c4a6a490c28f96b?pvs=4
- Needs https://github.com/reflex-dev/reflex/pull/2375 to compile properly

### CODE REVIEW

File: docs/library/datadisplay/icon.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+## Color
+
+```python demo
+flex(
+    icon(tag="magnifying_glass", color="indigo"),
+    icon(tag="magnifying_glass", color="cyan"),
+    icon(tag="magnifying_glass", color="orange"),
+    icon(tag="magnifying_glass", color="crimson"),
```
Comment (masenf) [inline docs/library/datadisplay/icon.md:81]: are these intended to be CSS colors? or radix themes colors?

i think at least one example should show `color="var(--accent-10)"` to link up with radix themes values.

**Review by masenf — CHANGES_REQUESTED** [review]

File: docs/library/datadisplay/icon.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+## List of Icons
+
+```python eval
+flex(
+    CalloutRoot.create(
+        CalloutIcon.create(icon(tag="check_circled", variant="solid", high_contrast=True, color="green")),
+        CalloutText.create("Below is a list of all available icons.", color="black", weight="bold"),
+        color="green",
+    ),
```
[the remark this answers is not in the document]
Comment (martinxu9) [reply docs/library/datadisplay/icon.md:33]: The creates do work. Not sure why I thought the snake cases were not defined and I needed to do creates 🤦🏼 I was using the flex, icon, etc just fine

File: docs/library/datadisplay/icon.md (hunk 1 above)
Comment (martinxu9) [reply docs/library/datadisplay/icon.md:81]: I think these are css colors? The component itself is supposedly primitive and does not take the common props. Just tried, the radix colors work, gonna add a couple examples there.

Comment (martinxu9) [reply docs/library/datadisplay/icon.md]: added in https://github.com/reflex-dev/reflex-web/pull/355/commits/b550f41698031201b83fddb19c301f041f3adfd0

File: docs/library/datadisplay/avatar.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    avatar(fallback="RX", variant="soft", high_contrast=True),
+    rows="2",
+    gap="2",
+    flow="column",
+)
+```
+
+### Radius
+
+We use the radius prop to assign a specific radius value, ignoring the global theme.
```
[the remark this answers is not in the document]
Comment (martinxu9) [reply docs/library/datadisplay/avatar.md]: Yes, adding them in https://github.com/reflex-dev/reflex-web/pull/355/commits/22d1acd2b4ef849f134688f761800061ee7e970a

File: docs/library/datadisplay/avatar.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    avatar(fallback="RX", color_scheme="cyan"),
+    avatar(fallback="RX", color_scheme="orange"),
+    avatar(fallback="RX", color_scheme="crimson"),
+    gap="2",
+)
+```
+
+### High Contrast
+
+We use the highContrast prop to increase color contrast of the fallback text with the background.
```
[the remark this answers is not in the document]
Comment (martinxu9) [reply docs/library/datadisplay/avatar.md]: oh .... thanks 🥲

**Comment by martinxu9** [comment]
revision uploaded! @masenf PTAL

File: docs/library/datadisplay/icon.md
Comment (Alek99) [inline docs/library/datadisplay/icon.md:1]: Can we break this up into sections like radix and include a link to radix icons at the top of the file as a source.

Comment (Alek99) [reply docs/library/datadisplay/icon.md:1]: Also may make them smaller and 4 or 5 width hard to scroll through all of them

**Review by Alek99 — CHANGES_REQUESTED** [review]

**Review by Alek99 — APPROVED** [review]

**Pull request merged**